lib/instrumentation/elasticsearch-shared.js

```diff
@@ -51,12 +44,18 @@ exports.setElasticsearchDbContext = function (span, path, query, body, isLegacy)
     if (body) {
       if (typeof (body) === 'string') {
         parts.push(body)
-      } else if (isLegacy) {
-        if (Array.isArray(body)) {
-          parts.push(body.map(JSON.stringify).join('\n')) // ndjson
-        } else if (typeof (body) === 'object') {
+      } else if (Buffer.isBuffer(body) || typeof body.pipe === 'function') {
+        // Never serialize a Buffer or a Readable. These guards mirror
+        // `shouldSerialize()` in the ES client, e.g.:
+        // https://github.com/elastic/elastic-transport-js/blob/069172506d1fcd544b23747d8c2d497bab053038/src/Transport.ts#L614-L618
+      } else if (Array.isArray(body)) {
+        try {
+          parts.push(body.map(JSON.stringify).join('\n') + '\n') // ndjson
+        } catch {}
+      } else if (typeof (body) === 'object') {
+        try {
           parts.push(JSON.stringify(body))
-        }
+        } catch {}
```

Comment:
REVIEW NOTE: try/catch around these in case there is a surprise circular ref or something. I'll be opening a separate ticket to improve our instrumentation patching so that we can capture the body and querystring *as serialized by the client* instead of having to do it (again) in the agent.